07.ls_object/ls.rb

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require_relative 'lib/screen'
+require_relative 'lib/ls_file_stat'
+
+def main
+  all_visible = false
+  reverse = false
+  long_format = false
+
+  opt = OptionParser.new
+  opt.on('-a') { all_visible = true }
+  opt.on('-r') { reverse = true }
+  opt.on('-l') { long_format = true }
+
+  paths = opt.parse(ARGV)
+  paths = ['.'] if paths.empty?
+
+  stats = LsFileStat.bulk_create(paths, reverse:)
+  file_stats = stats.filter(&:file?)
+  dir_stats = stats.filter(&:directory?)
```

Comment:
```suggestion
  file_stats, dir_stats = LsFileStat.bulk_create(paths, reverse:)
```

みたいに bulk_create 側でフィルタリングしてあげるのも一手（好みの問題です）

Reply:
もともとはそのようにしていたのですが、なんかちょっと違うかな、と思いやめた経緯があります・・・